MRIBase/src/Datatypes/Flags.jl

```diff
@@ -0,0 +1,96 @@
+export FLAGS, create_flag_bitmask, flag_is_set, flag_set!, flag_remove!, flag_remove_all!
+FLAGS = Dict(
+    "ACQ_FIRST_IN_ENCODE_STEP1"                => 1,
+    "ACQ_LAST_IN_ENCODE_STEP1"                 => 2,
+    "ACQ_FIRST_IN_ENCODE_STEP2"                => 3,
+    "ACQ_LAST_IN_ENCODE_STEP2"                 => 4,
+    "ACQ_FIRST_IN_AVERAGE"                     => 5,
+    "ACQ_LAST_IN_AVERAGE"                      => 6,
+    "ACQ_FIRST_IN_SLICE"                       => 7,
+    "ACQ_LAST_IN_SLICE"                        => 8,
+    "ACQ_FIRST_IN_CONTRAST"                    => 9,
+    "ACQ_LAST_IN_CONTRAST"                     => 10,
+    "ACQ_FIRST_IN_PHASE"                       => 11,
+    "ACQ_LAST_IN_PHASE"                        => 12,
+    "ACQ_FIRST_IN_REPETITION"                  => 13,
+    "ACQ_LAST_IN_REPETITION"                   => 14,
+    "ACQ_FIRST_IN_SET"                         => 15,
+    "ACQ_LAST_IN_SET"                          => 16,
+    "ACQ_FIRST_IN_SEGMENT"                     => 17,
+    "ACQ_LAST_IN_SEGMENT"                      => 18,
+    "ACQ_IS_NOISE_MEASUREMENT"                 => 19,
+    "ACQ_IS_PARALLEL_CALIBRATION"              => 20,
+    "ACQ_IS_PARALLEL_CALIBRATION_AND_IMAGING"  => 21,
+    "ACQ_IS_REVERSE"                           => 22,
+    "ACQ_IS_NAVIGATION_DATA"                   => 23,
+    "ACQ_IS_PHASECORR_DATA"                    => 24,
+    "ACQ_LAST_IN_MEASUREMENT"                  => 25,
+    "ACQ_IS_HPFEEDBACK_DATA"                   => 26,
+    "ACQ_IS_DUMMYSCAN_DATA"                    => 27,
+    "ACQ_IS_RTFEEDBACK_DATA"                   => 28,
+    "ACQ_IS_SURFACECOILCORRECTIONSCAN_DATA"    => 29,
+    "ACQ_COMPRESSION1"                         => 53,
+    "ACQ_COMPRESSION2"                         => 54,
+    "ACQ_COMPRESSION3"                         => 55,
+    "ACQ_COMPRESSION4"                         => 56,
+    "ACQ_USER1"                                => 57,
+    "ACQ_USER2"                                => 58,
+    "ACQ_USER3"                                => 59,
+    "ACQ_USER4"                                => 60,
+    "ACQ_USER5"                                => 61,
+    "ACQ_USER6"                                => 62,
+    "ACQ_USER7"                                => 63,
+    "ACQ_USER8"                                => 64
+)
+
+function bitshift(A, k)
+  if k >= 0
+      out = A << k
+  else
+      out = A >> abs(k)
+  end
+
+  return out
+end
+
+function create_flag_bitmask(flag)
+  # Determine the bit position for the flag
+  if isa(flag, String)
+    b = FLAGS[flag]
+  elseif flag > 0
+      b = UInt64(flag)
+      @info "Bitmask for "
+  else
+    @error "Flag is of the wrong type."
+  end
+  bitmask = bitshift(UInt64(1), b - 1)
+  return  bitmask
+end 
+
+function flag_is_set(obj::Profile, flag)
```

Comment:
Depending on how strictly you want to follow the Julia style guide, we could rename the functions to:

```julia
flag(obj::Profile, flag)
flag!(obj::Profile, flag, value::Boolean) # this would be both flag_set! and flag_remove!
```

and then maybe a helper function to remove everything

Reply:
Maybe it will be easier for users if we keep the same name as the python functions : 
https://github.com/ismrmrd/ismrmrd-python/blob/master/ismrmrd/flags.py

I don't know if we really need to add the matlab ones `setFlags` like they also did in the python package

Reply:
Ah, yes that sounds sensible!

Reply:
I'm adding this functions / docstrings and maybe some documentations after that I think we will be ready to merge :)

```julia
function flags_of(obj::Profile)
  flags_ = String[]
  for f in FLAGS
    flag_is_set(obj::Profile, f.first) ? push!(flags_,f.first) : nothing
  end
  return flags_
end
```